### PR TITLE
[Puppeteer] Use default tab instead of opening a new tab when starting the browser and new helper function I.grabNumberOfTabs

### LINKED ISSUE #908: open two tab with run multiple
#### What are you trying to achieve?
I would like open only one tab in chromedriver

#### What do you get instead?
https://i.imgur.com/nhEco2W.png

> codeceptjs run-multiple --all --reporter mocha-multi

```js
Feature('test1')
const scenario1 =
	'@test1|' +
	'bla bla bla 1'
Scenario(scenario1, function*(I) {
I.login('id')...
})
```

### Details

* CodeceptJS version:
* NodeJS Version:
* Operating System:
* Protractor || WebDriverIO || Nightmare version (if related)
* Configuration file:
"dependencies": {
"codeceptjs": "^1.1.3",
"concurrently": "^3.5.1",
"puppeteer": "^1.0.0"
},
"devDependencies": {
"mocha": "^4.1.0",
"mocha-multi": "^0.11.1",
"mocha-junit-reporter": "^1.15.0",
"mochawesome": "^3.0.1",
"mochawesome-report-generator": "^3.1.0"
}
```js
const multipleConfig= {
	test1: {
		grep: '@test1',
		browsers: ['chrome'],
	},
}
const defaultConfig = {
	tests: './src/tests/**/*.js',
	timeout: 900,
	output: './output',
	helpers: {
		Puppeteer: {
			windowSize: '1280x800',
			browser: 'chrome',
			chrome: {args: ['--no-sandbox', '--disable-setuid-sandbox']},
			url: 'http://localhost',
			uniqueScreenshotNames: false,
			show: false,
			waitForAction: 1000,
			restart: true,
		},
		MyHelpers: {
			require: './helpers/helpers.js',
		},
	},
	mocha: {
		reporterOptions: {
			'mocha-junit-reporter': {
				stdout: '-',
				options: {
					mochaFile: './output/result.xml',
				},
			},
			mochawesome: {
				stdout: './output/console.log',
				options: {
					reportDir: './output',
					reportPageTitle: 'Report page',
					reportFilename: 'report',
					quiet: true,
					json: false,
					html: false,
					overwrite: true,
					inline: true,
				},
			},
		},
	},
	multiple: multipleConfig,
	bootstrap: false,
	include: {},
	name: 'codecept',
}
```

### PR DESCRIPTION
**Pull request by reubenmiller**
### New Helper Functions
[Puppeteer|WebDriverIO|Protractor]
* I.grabNumberOfTabs: Returns the number of tabs which are currently opened in the browser. Useful for checking that no additional tabs have been opened, or to confirm the closing of tabs.

### [Puppeteer] Use default tab instead of opening a new one when starting the browser
* Fixes #908 : A fairly obvious fix. Added tests to check for this in the future (hence the need for the new helper function `I.grabNumberOfTabs`)

### CODE REVIEW

**Comment by reubenmiller** [comment]
Sometimes the `beforeEach` hook is failing on the WebDriverIO tests. I have a feeling the test runner is sometimes flakey/unresponsive and the test timeout triggers.

@DavertMik Can you trigger the travis-ci tests again, or are you ok with a false positive test fail?

File: lib/helper/Protractor.js [hunk 1]
```diff
@@ -874,6 +874,18 @@ class Protractor extends Helper {
     return client.switchTo().window(newHandle);
   }
 
+  /**
+   * Grab number of open tabs
+   *
+   * ```js
+   * I.grabNumberOfOpenTabs();
```
Comment (DavertMik) [inline lib/helper/Protractor.js]: This function is used across different helpers so I'd recommend to move the documentation to external file and include it. 

Create `grabNumberOfOpenTabs.md` inside https://github.com/Codeception/CodeceptJS/tree/master/docs/webapi

Place documentation there. Include this files here as 

```
{{> ../webapi/grabNumberOfOpenTabs }}
```
Also for grabbers, please provide an example with the variable actually assigned:

```js
const tabsNum = await I.grabNumberOfOpenTabs();
```

Comment (reubenmiller) [reply lib/helper/Protractor.js]: Sure makes sense. Though I assumed you meant a `.mustache` file instead of `.md`.

**Comment by reubenmiller** [comment]
The tests are a bit of a pain...I'm not sure if it is a problem with the resources on travis-ci or if we have some problems some test cases/helpers. It always seems to be a different test which fails, and mostly nothing to do with the PR :/

**Comment by DavertMik** [comment]
Ok, I restarted some builds and tests passed. Thanks!

**Pull request merged**